internal/controller/authenticator/jwtcachefiller/jwtcachefiller.go

```diff
@@ -210,16 +299,194 @@ func newJWTAuthenticator(spec *auth1alpha1.JWTAuthenticatorSpec) (*jwtAuthentica
+
+func (c *jwtCacheFillerController) validateProviderJWKSURL(provider *coreosoidc.Provider, pJSON *providerJSON, conditions []*metav1.Condition, prereqOk bool) (string, []*metav1.Condition, error) {
+	if provider == nil || pJSON == nil || !prereqOk {
+		conditions = append(conditions, &metav1.Condition{
+			Type:    typeJWKSURLValid,
+			Status:  metav1.ConditionUnknown,
+			Reason:  reasonUnableToValidate,
+			Message: msgUnableToValidate,
+		})
+		return "", conditions, nil
+	}
+	// should be impossible because coreosoidc.NewProvider validates this, thus we can't write a test to get in this state (currently)
+	if err := provider.Claims(pJSON); err != nil {
+		errText := "could not get provider jwks_uri"
+		msg := fmt.Sprintf("%s: %s", errText, err.Error())
+		conditions = append(conditions, &metav1.Condition{
+			Type:    typeJWKSURLValid,
+			Status:  metav1.ConditionFalse,
+			Reason:  reasonInvalidProviderJWKSURL,
+			Message: msg,
+		})
+		// resync err, the user may not be able to fix this via config, it may be the server may be misbehaving.
+		return pJSON.JWKSURL, conditions, fmt.Errorf("%s: %w", errText, err)
+	}
+
+	parsedJWKSURL, err := url.Parse(pJSON.JWKSURL)
+	if err != nil {
+		errText := "could not parse provider jwks_uri"
+		msg := fmt.Sprintf("%s: %s", errText, err.Error())
+		conditions = append(conditions, &metav1.Condition{
+			Type:    typeJWKSURLValid,
+			Status:  metav1.ConditionFalse,
+			Reason:  reasonInvalidProviderJWKSURL,
+			Message: msg,
+		})
+		// resync err, the user may not be able to fix this via config, it may be the server may be misbehaving.
+		return pJSON.JWKSURL, conditions, fmt.Errorf("%s: %w", errText, err)
+	}
+
+	// spec asserts https is required. https://openid.net/specs/openid-connect-discovery-1_0.html
+	if parsedJWKSURL.Scheme != "https" {
+		msg := fmt.Sprintf("jwks_uri %s has invalid scheme, require 'https'", pJSON.JWKSURL)
+		conditions = append(conditions, &metav1.Condition{
+			Type:    typeJWKSURLValid,
+			Status:  metav1.ConditionFalse,
+			Reason:  reasonInvalidProviderJWKSURLScheme,
+			Message: msg,
+		})
+		return pJSON.JWKSURL, conditions, fmt.Errorf("%s", msg)
+	}
+
+	conditions = append(conditions, &metav1.Condition{
+		Type:    typeJWKSURLValid,
+		Status:  metav1.ConditionTrue,
+		Reason:  reasonSuccess,
+		Message: "jwks_uri is a valid URL",
+	})
+	return pJSON.JWKSURL, conditions, nil
+}
+
+func (c *jwtCacheFillerController) validateTLS(spec *auth1alpha1.JWTAuthenticatorSpec, conditions []*metav1.Condition) (*x509.CertPool, []*metav1.Condition, bool) {
+	rootCAs, _, err := pinnipedauthenticator.CABundle(spec.TLS)
+	if err != nil {
+		msg := fmt.Sprintf("%s: %s", "invalid TLS configuration", err.Error())
+		conditions = append(conditions, &metav1.Condition{
+			Type:    typeTLSConfigurationValid,
+			Status:  metav1.ConditionFalse,
+			Reason:  reasonInvalidTLSConfiguration,
+			Message: msg,
+		})
+		return rootCAs, conditions, false
+	}
+
+	msg := "valid TLS configuration"
```

Comment:
These success messages would be more clear if they said what was valid. Otherwise, they are just repeating the type name of the condition and the user has no insight into what validations were performed. For example, this one could say something like `Successfully parsed specified CA bundle` if it parsed one, or `No CA bundle specified` if it did not parse one. Both are happy cases. Or if you don't want to have different messages then perhaps combine both happy cases into one message something like `Parsed CA bundle, if one was specified`.